src/go/k8s/tests/e2e-v2/upgrade-values-check/01-upgrade-redpanda.yaml

```diff
@@ -0,0 +1,37 @@
+---
+apiVersion: cluster.redpanda.com/v1alpha1
+kind: Redpanda
+metadata:
+  name: redpanda
+spec:
+  chartRef:
+    chartVersion: "5.0.9"
+  clusterSpec:
+    statefulset:
+      replicas: 1
+    tls:
+      enabled: false
+    auth:
+      sasl:
+        enabled: true
+        users:
+          - name: superuser
+            password: secretpassword
+    listeners:
+      admin:
+        port: 9640
+        external:
+          default:
+            port: 9645
+            tls:
+              requireClientAuth: false
+              cert: external
+            advertisedPorts:
+              - 31644
+        tls:
+          requireClientAuth: false
+          enabled: false
+    storage:
+      persistentVolume:
+        enabled: true
+        size: 20Gi
```

Comment:
```suggestion
apiVersion: cluster.redpanda.com/v1alpha1
kind: Redpanda
metadata:
  name: redpanda
spec:
    listeners:
      admin:
        port: 9644
```

kuttle steps can easily reconcile the end state. Using this form it's clear that port has changed.

Reply:
Its because i want to know that it does reconcile, not I am not testing the end state, just that it does reconcile and the end status is as expected. I have used a new function to detect variances, i just want to make sure it works as expected.

Reply:
> Its because i want to know that it does reconcile, not I am not testing the end state, just that it does reconcile and the end status is as expected.

I don't understand.

> I have used a new function to detect variances, i just want to make sure it works as expected.

I understand that. I'm not telling you to change the test case. This will do exactly what you want. The upgrade (actual change) is done only on the admin listener port. kuttle engine will do the merge patching and you will end up exactly with the same Redpanda resource. For maintainer point of view it's much easier to understand what the step 01 was trying to achieve. From the initial 00-create-redpanda step it only change the Admin listener port. It's much easier to comprehend.

See how docs are describing writing second step
https://kuttl.dev/docs/kuttl-test-harness.html#write-a-second-test-step

And another reference for updating existing object
https://kuttl.dev/docs/testing/steps.html#updating-objects

Reply:
> > Its because i want to know that it does reconcile, not I am not testing the end state, just that it does reconcile and the end status is as expected.
> 
> I don't understand.
> 
> > I have used a new function to detect variances, i just want to make sure it works as expected.
> 
> I understand that. I'm not telling you to change the test case. This will do exactly what you want. The upgrade (actual change) is done only on the admin listener port. kuttle engine will do the merge patching and you will end up exactly with the same Redpanda resource. For maintainer point of view it's much easier to understand what the step 01 was trying to achieve. From the initial 00-create-redpanda step it only change the Admin listener port. It's much easier to comprehend.
> 
> See how docs are describing writing second step https://kuttl.dev/docs/kuttl-test-harness.html#write-a-second-test-step
> 
> And another reference for updating existing object https://kuttl.dev/docs/testing/steps.html#updating-objects

I understand what you were trying to say now. Yes, you are right this should just be doing patches as how kuttl works. I keep forgetting that since it seems like we should be specifiying what we want the resource to look like, this is confusing, thank you again for pointing me to the right direction